lms/djangoapps/oauth_dispatch/tests/test_views.py

```diff
@@ -153,43 +153,78 @@ class TestAuthorizationView(TestCase):
     def setUp(self):
         super(TestAuthorizationView, self).setUp()
         self.user = UserFactory()
-        self.dop_client = self._create_confidential_client(user=self.user, client_id='dop-app-client-id')
-
-    def _create_confidential_client(self, user, client_id):
-        """
-        Create a confidential client suitable for testing purposes.
-        """
-        return self.dop_adapter.create_confidential_client(
-            name='test_app',
-            user=user,
-            client_id=client_id,
-            redirect_uri=DUMMY_REDIRECT_URL
+        self.dot_app = self.dot_adapter.create_confidential_client(
+            name='test dot application',
+            user=self.user,
+            redirect_uri=DUMMY_REDIRECT_URL,
+            client_id='confidential-dot-app-client-id',
+        )
+        self.dop_app = self.dop_adapter.create_confidential_client(
+            name='test dop client',
+            user=self.user,
+            redirect_uri=DUMMY_REDIRECT_URL,
+            client_id='confidential-dop-app-client-id',
         )
 
-    def test_authorization_view(self):
+    @ddt.data('dop', 'dot')
+    def test_post_authorization_view(self, client_type):
+        oauth_application = getattr(self, '{}_app'.format(client_type))
         self.client.login(username=self.user.username, password='test')
+        allow_field = {'dop': 'authorize', 'dot': 'allow'}
```

Comment:
Nit (feel free to ignore): for readability, consider putting 'authorize' and 'allow' inside tuples in the `ddt`.